controllers/component_controller.go

```diff
@@ -328,6 +329,40 @@ func (r *ComponentReconciler) runBuild(ctx context.Context, component *appstudio
 
 	log := r.Log.WithValues("Namespace", component.Namespace, "Application", component.Spec.Application, "Component", component.Name)
 
+	// Not exposing this as config on purpose since this is supposed to be moved out eventually
+	// Also, this is really being used for tests.
+	credentialPath := os.Getenv("BUILD_CREDENTIALS")
+	if credentialPath == "" {
+		// expect this to be unset in production.
+		credentialPath = "/etc/build/.dockerconfigjson"
+	}
+
+	if strings.Contains(component.Spec.Build.ContainerImage, r.ImageRepository) {
```

Comment:
Is it worth moving the code into a separate function? (maybe include the block above too)

Reply:
Yes 👍